lexic.txt

```diff
@@ -0,0 +1,50 @@
+Student: Condrea Adrian
+Lab 1b
+Lexic.txt
+
+Specification of the programming language
+
+Alphabet:
+letter = (“a”, “b”, …, “z”, “A”, “B”, …, “Z”);
+underline =  ‘_’;
+digit = (“0”, “1”, …, “9”);
+
+The lexic of the programming language contains:
+
+Special symbols:
+operator = (“+”, “-“, “*”, “/“, “<=“, “>=“, “==“, “!=“, “<“, “>”, “%”, “!”);
+separator = (“[“, “]”, “(“, “)”, “<“, “>”, “,”, “\””, “’”, “ “, “\n”, “.”);
+
+Reserved words:
+reservedWords = “startProgram”, “input”, “declare”, “typeCheck”, “typeDefine”, “check”, “else”, “output”, “endProgram”, “assign”, “loop”, “breakLoop”, “integer”, “boolean”, “string”, “char”, “array”;
+
+Identifiers
+Are a sequence of letters and digits. 
+
+Identifier rules:
+An identifier consists of a letter or a sequence of any letters, digits, or underline 
+The first character is always a letter
+The number of characters is <= 256
+Identifiers can not be the same as reserved words
+
+identifier = letter | letter, {( letter, digit, underline)};
+letter = (“a”, “b”, …, “z”, “A”, “B”, …, “Z”);
+digit = (“0”, “1”, …, “9”);
+underline = “_”;
+
+Constants
+Integer - rule:
+noConst = “0” | (“+”, “-“),digit - “0”, {digit};
```

Comment:
If you want to use a non-zero digit I think you should define it and use that instead of `digit - “0”`, I think that is a little bit misleading and  not sure if it's correct.
Also, with your current rule you need to specify the sign of the integer even if it is a positive one thus referring to 1 is invalid but +1 is correct. I think you should use ["-"] and specify the sign only for negative numbers.

Reply:
Thank you, good observations :). I have updated the file